integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -333,4 +440,276 @@ describe("Dictionary", () => {
+      //             TestCase.assertTrue(obj.x2 !== undefined,"This field should be equal x2");
+      //             TestCase.assertTrue(obj.x1 !== undefined,"This field should be equal x1");
+      //             TestCase.assertArrayLength(changeset.deletions, 3, "deletions");
+      //             TestCase.assertArrayLength(changeset.insertions, 2, "insertions");
+      //             TestCase.assertArrayLength(changeset.modifications, 0, "modifications");
+      //         }
+      //         if (cnt === 2) {
+      //             TestCase.assertTrue(obj.x1 !== undefined,"This field should be equal x1");
+      //             TestCase.assertTrue(obj.x5 !== undefined,"This field should be equal x5");
+      //             TestCase.assertTrue(obj.x3 !== undefined,"This field should be equal x3");
+      //             TestCase.assertArrayLength(changeset.deletions, 2, "deletions");
+      //             TestCase.assertArrayLength(changeset.insertions, 3, "insertions");
+      //             TestCase.assertArrayLength(changeset.modifications, 0, "modifications");
+      //         }
+      //         if (cnt === 3) {
+      //             let keys = Object.keys(obj);
+      //             TestCase.assertEqual(keys[0], "x1", "First field should be equal x1");
+      //             TestCase.assertEqual(obj.x1, "hello", "x1 should be equals to \"hello\"");
+      //             TestCase.assertArrayLength(changeset.deletions, 3, "deletions");
+      //             TestCase.assertArrayLength(changeset.insertions, 1, "insertions");
+      //             TestCase.assertArrayLength(changeset.modifications, 0, "modifications");
+      //         }
+      //         cnt++;
+      //     }
+      //     ff.fields.addListener(a);
+      //     // total object mutation.
+      //     realm.write(() => { ff.fields = {x1: 1, x2: 2} } );
+      //     // partial object mutation.
+      //     realm.write(() => { ff.fields = {x1: 1, x3: 2, x5: 5} } );
+      //     // deleting all but one field.
+      //     realm.write(() => { ff.fields = {x1: "hello"} } );
+      //     return new Promise((resolve, reject) => {
+      //         setTimeout(() => {
+      //             TestCase.assertEqual(cnt, 4, "Counter should be four");
+      //             ff.fields.removeAllListeners();
+      //             realm.close();
+      //             resolve();
+      //         }, 1000);
+      //     })
+    });
+    it("support removeAll notifications", function (this: RealmContext) {
+      this.realm.write(() => this.realm.create(DictSchema.name, { fields: { x: 1, y: 2, z: 3 } }));
+      const point = this.realm.objects<IDictSchema>(DictSchema.name)[0].fields;
+
+      let cnt = 0;
+      for (let i = 0; i < 10; i++) {
+        point.addListener(() => {
+          cnt++;
+        });
+      }
+
+      point.removeAllListeners();
+      this.realm.write(() => (point.x = 10));
+
+      return new Promise<void>((resolve, reject) => {
+        setTimeout(() => {
```

Comment:
I have squigglies here saying that it can't fund `setTimeout()`, but I'm given multiple import options and I can't figure out which one we want
![image](https://user-images.githubusercontent.com/44238293/212350796-043596eb-bc08-459b-8ac6-bb60dc8436bd.png)
![image](https://user-images.githubusercontent.com/44238293/212350892-e9c0f9b7-b707-4e3c-a71e-027392c1c3e9.png)